From 2229f3b9fd4bad47794c28e558ad273173cea73d Mon Sep 17 00:00:00 2001
From: Andrea Adami <andrea.adami@gmail.com>
Date: Sat, 27 Jan 2018 09:52:46 +0100
Subject: [PATCH 9/9] Eliminate warnings about implicit non-const casting in
 libmtd

The mtd_get_dev_info1 function reads (among other things) name and type
string into coresponding struct mtd_dev_info fields.

The struct mtd_dev_info has the string fields marked const, requiring
them to be cast to non-const version during initialization.

This cast was previously omitted from the dev_read_data calls,
triggering warnings during compilation.

Signed-off-by: David Oberhollenzer <david.oberhollenzer@sigma-star.at>
Signed-off-by: Richard Weinberger <richard@nod.at>

Upstream-Status: Backport

Signed-off-by: Andrea Adami <andrea.adami@gmail.com>
---
 lib/libmtd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/libmtd.c b/lib/libmtd.c
index 60b4782..5f0bcbc 100644
--- a/lib/libmtd.c
+++ b/lib/libmtd.c
@@ -746,13 +746,13 @@ int mtd_get_dev_info1(libmtd_t desc, int mtd_num, struct mtd_dev_info *mtd)
 	if (dev_get_major(lib, mtd_num, &mtd->major, &mtd->minor))
 		return -1;
 
-	ret = dev_read_data(lib->mtd_name, mtd_num, &mtd->name,
+	ret = dev_read_data(lib->mtd_name, mtd_num, (char *)&mtd->name,
 			    MTD_NAME_MAX + 1);
 	if (ret < 0)
 		return -1;
 	((char *)mtd->name)[ret - 1] = '\0';
 
-	ret = dev_read_data(lib->mtd_type, mtd_num, &mtd->type_str,
+	ret = dev_read_data(lib->mtd_type, mtd_num, (char *)&mtd->type_str,
 			    MTD_TYPE_MAX + 1);
 	if (ret < 0)
 		return -1;
-- 
2.7.4

